### PR TITLE
Show delete run-path dialog only for non-empty iter

### PR DESCRIPTION
**Pull request by dafeda**
Resolves: #7633 

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/run_models/base_run_model.py [hunk 1]
```diff
@@ -613,6 +613,15 @@ def check_if_runpath_exists(self) -> bool:
         """
         return any(Path(run_path).exists() for run_path in self.paths)
 
+    def check_if_runpath_contains_data(self, iteration: int) -> bool:
+        target_folder = f"iter-{iteration}"
```
Comment (oyvindeide) [inline src/ert/run_models/base_run_model.py:617]: There is no requirement to have `iter` in the runpath, and for example in prediction runs it is not used, so changing `poly.ert` to: `RUNPATH poly_out/realization-<IENS>/<PRED_DIR>` is valid.

File: src/ert/run_models/base_run_model.py [hunk 2]
```diff
@@ -613,6 +613,15 @@ def check_if_runpath_exists(self) -> bool:
...
+
+        for run_path in self.paths:
```
Comment (oyvindeide) [inline src/ert/run_models/base_run_model.py:619]: Think perhaps this is the root cause of the problems, `self.paths` should only return what is actually going to be run in, but suspect there is a bug there.

File: src/ert/run_models/base_run_model.py (hunk 1 above)
Comment (dafeda) [reply src/ert/run_models/base_run_model.py:617]: Ah, that makes sense. Need to rethink this then.

**Comment by oyvindeide** [comment]
Suggest removing: `start_iteration` from `SimulationArguments` (as it is always 0), and letting the run model decide what the value is, for most it will be 0, but for ES-MDA it might be something else if it is a restart run.

**Comment by dafeda** [comment]
Solved by: https://github.com/equinor/ert/pull/7940

**Pull request closed without merging**